src/commands/synthetics/__tests__/cli.test.ts

```diff
@@ -164,6 +179,91 @@ describe('run-test', () => {
       })
     })
 
+    test('override from global < ENV < test file', async () => {
```

Comment:
I'm not sure to follow how this test works, so maybe we could add a few more comments?
Also the test just above seems to have a title going the other way: `override from config file < ENV < CLI`

Reply:
The test just above doesn't test the same code path.

- `test('override from config file < ENV < CLI')` is testing that CLI arguments have precedence over what's configured in the "config file" (e.g. `datadog-ci.json`)
- `test('override from global < ENV < test file')` is testing that what's configured in "test files" (e.g. tests.synthetics.json) has precedence over what's configured in the [`global` property](https://github.com/DataDog/datadog-ci/blob/bd0a8536ab8ab0bbf0bbd2d0136c944f3eba898b/.github/workflows/e2e/global.config.json#L5) of the "config file".

Reply:
After discussing offline, we concluded the best course of action was to:
- explain clearly the 6 different cases of overrides from config file to ENV to CLI to test file
- refactor the testing of these overrides in the test below (`override locations with ENV variable`) ...
- ... to later split it into smaller tests.